clients/src/main/java/org/apache/kafka/common/network/KafkaChannel.java

```diff
@@ -164,6 +169,24 @@ public Send write() throws IOException {
         return result;
     }
 
+    /**
+     * Accumulates network thread time for this channel.
+     */
+    public void addNetworkThreadTimeNanos(long nanos) {
+        networkThreadTimeNanos.addAndGet(nanos);
+    }
+
+    /**
+     * Returns accumulated network thread time for this channel.
+     * Broker resets the value after recording time spent on each request.
+     * <p>
+     * This is an AtomicLong because the value is updated by the network thread
+     * and read/reset by the broker's request handler thread.
+     */
+    public AtomicLong networkThreadTimeNanos() {
+        return networkThreadTimeNanos;
```

Comment:
AtomicLong is returned so that the the value can be updated from the I/O thread when a request is complete without propagating `KafkaChannel` to the request handling code.